lib/src/async/retry.dart

```diff
@@ -21,13 +21,22 @@ const _defaultTimeout = const Duration(seconds: 60);
 /// reached.
 ///
 /// Waits [interval] to perform the retry after a failed attempt.
+///
+/// If the task fails to complete once within the timeout it will resolve as an
+/// error with a TimeoutException. If the task completes with an error at least
+/// once before the timeout and never succeeds it will resolve as an error with
+/// the last exception from a task run that did not time out.
```

Comment:
I'm torn here on this contract. Does it make more sense / remain simple to just throw a timeout exception and not cache the last exception?